lib/hyrax/transactions/transaction.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+require 'dry/monads'
+require 'dry/monads/do'
+
+module Hyrax
+  module Transactions
+    ##
+    # @since 3.0.0
+    class Transaction
+      include Dry::Monads[:result]
+      include Dry::Monads::Do.for(:call)
+
+      ##
+      # @!attribute [rw] container
+      #   @return [Container]
+      # @!attribute [rw] steps
+      #   @return [Array<String>]
+      attr_accessor :container, :steps
+
+      ##
+      # @param [Container] container
+      # @param [Array<String>] steps
+      def initialize(container: Container, steps:)
+        self.container = container
+        self.steps     = steps
+      end
+
+      ##
+      # @param [Valkyrie::ChangeSet] change_set
+      #
+      # @return [Dry::Monads::Result]
+      def call(change_set)
```

Comment:
Can I get a comment on what is happening here? Perhaps expand the variable naming of `w` and `s`.

My understanding of `#inject` is that the yielded value will become the `w` for the next `s`.

Reply:
will rename the variables for clarity, and add a top-level method comment about what this is doing.

Reply:
maybe the rename and added docs help?